src/reconcile/rec_write.c

```diff
@@ -1488,7 +1491,11 @@ __rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
 			    WT_TXNID_LT(upd->txnid, r->unstable_txn))
 				r->unstable_txn = upd->txnid;
 			if (upd->start_ts < r->unstable_timestamp)
-				r->unstable_timestamp = upd->start_ts;
+				r->unstable_timestamp =
+				r->unstable_durable_timestamp = upd->start_ts;
+			if (upd->durable_ts < r->unstable_durable_timestamp &&
+				upd->durable_ts > r->unstable_timestamp)
```

Comment:
I don't understand the second part of the clause here, I tried the following instead and it passed the tests I tried with it:

```
                        /*                                                      
                         * The durable timestamp is always set, and usually     
                         * the same as the start timestamp, which makes it OK   
                         * to use the two independently and be confident both   
                         * will be set.                                         
                         */                                                     
                        WT_ASSERT(session, upd->durable_ts >= upd->start_ts);   
                        if (upd->start_ts < r->unstable_timestamp)              
                                r->unstable_timestamp = upd->start_ts;          
                        if (upd->durable_ts < r->unstable_durable_timestamp)    
                                r->unstable_durable_timestamp = upd->durable_ts;
```

Reply:
As discussed, the suggested assertion would fail for updates with prepare in-progress state. So I have further augmented this assertion to check for the prepare state the update.